### DISCUSSION TITLE
Recommended way to find a filename prior to polling it for the defpoll variable?

### DISCUSSION THREAD

**Discussion by Animeshz** [elkowar/eww#314] [Q&A] [answered]
As a general way it could be done each iteration in the script, but I feel there could be another way right?

I think I should export some env variable or pass some arguments when starting eww itself, is there way to pass as such args? Or something like one-time poll within eww because it might be a bit clunky to pass parameter every time using the eww command?

**Comment by elkowar** [comment, marked as answer]
I don't think there is currently any easy way to do this in pure eww. However, given that script-vars are just shell scripts, I _think_ they should be able to read any environment variables that are available when the eww daemon is launched. Thus, you should be able to export your variables as environment variables for the eww daemon, and then just refer to them as normal env vars in your yuck configuration scripts.

Haven't actually tried this, but I don't see any reason for it not to work.

**Reply by Animeshz** [reply]
Yeah, seems good way to export them on the rc files, and just refer or have a fallback :))
